.github/workflows/unit-tests.yml

```diff
@@ -7,9 +7,9 @@ jobs:
     strategy:
       matrix:
         startarg:
-          - { make: 'check-memory', cc: 'clang', tag: 'i386' }
+          - { make: 'check-memory', cc: 'clang', tag: 'latest' }
           - { make: 'check-memory', cc: 'gcc', tag: 'latest' }
-          - { make: 'distcheck', cc: 'clang', tag: 'latest' }
+          - { make: 'distcheck', cc: 'clang', tag: 'i386' }
           - { make: 'distcheck', cc: 'gcc', tag: 'i386' }
```

Comment:
why is distcheck moved to i386? Is it because check-memory also does it so it would overlap?

Reply:
I'd still like to run `make check` on both amd64 and i386. Sure we can probably reduce one of them to `check`, as the "dist" part of "distcheck" isn't really architecture specific. So that would be an optimization, but not sure if it's really worth it. We didn't mind running it twice before either.

Reply:
I think the pairwise isn't quite right here anymore.

I think it's weird that we only run distcheck on an architecture that nobody uses.  I also don't see the value of running distcheck with separate compilers.  I see a very very minor potential value in having valgrind with different compilers.

So I'd say either:

 - amd64 `check-memory` with gcc and/or clang (and/or = depending how much we care)
 - `distcheck` on amd64 with some compiler
 - `distcheck` on i386 with the other compiler

Reply:
My thinking was that things that break the `dist` in `distcheck` are not architecture specific. I want to continue testing {amd64, i386} x {gcc, clang}, so we need at least four scenarios. I also actually do care about valgrinding both gcc and clang.

So if you care about running distcheck on amd64, we then need to run `make check-memory distcheck` on one of them, which is certainly possible -- it will just take a bit longer and break the symmetry.

Good point about the comment, I'll add that once we have an agreement how to shuffle the scenarios.

Reply:
Shuffled, WDYT?

I'm not sure about whether the `--make` option will work with multiple targets, I'll find out.

Reply:
Reworked as discussed.